project/First-Project/src/main/java/first/project/service/AuthenticationService.java

```diff
@@ -0,0 +1,18 @@
+package first.project.service;
+
+import first.project.model.User;
+
+import java.util.ArrayList;
+
+public interface AuthenticationService
+{
+    boolean checkEmail(String email);
+
+    boolean checkUserName(String username);
+
+    boolean checkPassword(String password);
+
+    void passwordEncode(User user);
```

Comment:
You don't need separate passwordEncode function in the service. If you need this function, you need to create it as a private function in the implementation of the service